### DISCUSSION TITLE
[Enhancement] MAUI Controls Gallery

### DISCUSSION THREAD

**Discussion by LeftTwixWand** [dotnet/maui#183] [General]
## Create open source MAUI Controls Gallery
I think it'll be useful to create MAUI Controls Gallery, similar to [XAML Controls Gallery](https://github.com/microsoft/Xaml-Controls-Gallery).

It's difficult to start with new technologies for new developers. And projects, like controls gallery are incredible useful for that.

**Comment by davidortinau** [comment]
Let's do it! Let's discuss.

What makes for a really good control gallery?

**Reply by hbcondo** [reply]
I think these are really good control galleries:

- https://gallery.platform.uno
- https://gallery.flutter.dev
- https://getbootstrap.com/docs/5.0/components

**Comment by LeftTwixWand** [comment]
If MAUI is really multi-platform, we can create similar application for Windows(using Project Reunion), Android and IOS.
I think, that XAML Controls Gallery is a good base for new gallery. But we should add code switcher for developers, who prefer to work with 

MVVM:
```XAML
<StackLayout>
    <Label Text="Welcome to .NET MAUI!" />
    <Button Text="{Binding Text}" 
            Command="{Binding ClickCommand}" />
</StackLayout>
```
And MVU:
```csharp
[Body]
View body() => new StackLayout
{
    new Label("Welcome to .NET MAUI!"),
    new Button(
        () => $"You clicked {count} times.",
        () => count.Value ++)
    )
};
```
And I also think, it will be cool to create few quality design templates. For example: To-Do List, Weather Forecast, News application.

**Comment by zprima** [comment]
Having some app samples that use the controls in various scenarios would also be helpful. Especially for new developers, to see how they all work together. Something similar that Compose has https://github.com/android/compose-samples

**Comment by RedEye-Developers** [comment]
any updates about MAUI Controls Gallery same like winui 3 gallary ?
